bitnami/kong/README.md

```diff
@@ -215,6 +215,7 @@ The following tables list the configurable parameters of the kong chart and thei
 | `metrics.service.annotations`                   | Annotations for Prometheus metrics service                                                                                                            | `Check values.yaml file`                                                                                    |
 | `metrics.serviceMonitor.enabled`                | if `true`, creates a Prometheus Operator ServiceMonitor (also requires `metrics.enabled` to be `true`)                                                | `false`                                                                                                     |
 | `metrics.serviceMonitor.namespace`              | Namespace in which Prometheus is running                                                                                                              | `nil`                                                                                                       |
+| `metrics.serviceMonitor.serviceAccount`         | Service account used by Prometheus                                                                                                                    | `nil`
```

Comment:
Could we follow used on other charts (e.g. MariaDB Galera or Kafka) where we have 3 different parameters: `serviceAccount.create`, `serviceAccount.name`, and `rbac.create`. This approach allow us to use a custom ServiceAccount without enabling RBAC.

Check the link below:

- https://github.com/bitnami/charts/tree/master/bitnami/kafka#rbac-parameters

Reply:
I am not sure if we should make it like the other charts, the idea is to use an existing `serviceAccount` which will be preconfigured for prometheus.

We you decide to integrate kong with prometheus, you have to write the namespace where prometheus is running. and write the serviceAccount which is used by prometheus.

I guess the creation would be only in case you need to create a serviceAccount for prometheus itself but this would be out of this chart's scope.

Am I getting what you meant correctly?

Reply:
Yes, you want to provide Prometheus the permissions to access the Kong services and gather the metrics. As you said, it's a different use case.

That said, you are assuming that Role,Rolebinding must be created when `metrics.enabled` and `metrics.serviceMonitor.enabled` are `true`. And that's not true on clusters where RBAC is not enabled.

That's why I still think it's necessary a `rbac.create` parameter to decide whether create these resources or not.

Reply:
I got it, I added the `rbac.create` condition which will create the resources if needed (enabled by default because we have another ingress rbac condition in this chart which is enabled by default too)